messaging/src/main/java/org/axonframework/messaging/responsetypes/MultipleInstancesResponseType.java

```diff
@@ -44,42 +44,88 @@
 public class MultipleInstancesResponseType<R> extends AbstractResponseType<List<R>> {
 
     private static final Logger logger = LoggerFactory.getLogger(MultipleInstancesResponseType.class);
+    private final InstanceResponseType<R> instanceResponseType;
 
     /**
-     * Instantiate a {@link MultipleInstancesResponseType} with the given
-     * {@code expectedCollectionGenericType} as the type to be matched against and which the convert function will use
-     * as the generic for the {@link java.util.List} return value.
+     * Instantiate a {@link MultipleInstancesResponseType} with the given {@code expectedCollectionGenericType} as the
+     * type to be matched against and which the convert function will use as the generic for the {@link java.util.List}
+     * return value.
      *
      * @param expectedCollectionGenericType the response type which is expected to be matched against and returned
      */
     @JsonCreator
     @ConstructorProperties({"expectedResponseType"})
     public MultipleInstancesResponseType(@JsonProperty("expectedResponseType") Class<R> expectedCollectionGenericType) {
         super(expectedCollectionGenericType);
+        instanceResponseType = new InstanceResponseType<>(expectedCollectionGenericType);
     }
 
+
     /**
-     * Match the query handler its response {@link java.lang.reflect.Type} with this implementation its responseType
-     * {@code R}.
-     * Will return true in the following scenarios:
+     * Match the query handler its response {@link Type} with this implementation its responseType {@code R}. Will
+     * return a value greater than 0 in the following scenarios:
```

Comment:
Curious: Why add "greater than 0" here?

Reply:
Not cleaned up well after refactor. Updated now.